### ISSUE TITLE
[BUG] `no-useless-await` is applied to locator methods regardless of receiver object

### ISSUE THREAD

**Issue by ppath** [playwright-community/eslint-plugin-playwright#350]
For the locator methods, the check is only against the method name and not the receiver object type. Therefore, in the following example the `await` before `a.locator();` is marked as useless:

```typescript
class A {
  locator(): Promise<void>;
}
async function () {
  const a = new A;
  await a.locator();
}
```

It seems the relevant logic is [here,](https://github.com/playwright-community/eslint-plugin-playwright/blob/6baec3ac2861b6cd2c8fcb83c61d00b4e3c82128/src/rules/no-useless-await.ts#L74) and I wonder if there could be an additional check on the receiver object for locator methods, like the one for page methods.

**Comment by mskelton** [comment]
This is intentional. We prefer false positives that can be ignored than false negatives that go uncaught.
